model/publishing/delivery/tasks/DeployTestEnvironments.php

```diff
@@ -160,18 +156,38 @@ private function getQtiTestPackageFile(core_kernel_classes_Resource $delivery):
     {
         $packagePath = $delivery->getOnePropertyValue($this->getProperty(TestBackupService::PROPERTY_QTI_TEST_BACKUP_PATH));
         $packagePath = (string)  $packagePath;
-        
+
         return $this->getServiceLocator()
             ->get(FileSystemService::SERVICE_ID)
             ->getDirectory(TestBackupService::FILESYSTEM_ID)
             ->getFile($packagePath);
     }
 
+    private function getDeliveryPropertyList(\core_kernel_classes_Resource $delivery): array
+    {
+        $propertyList = [];
+        foreach ($this->getPublishingDeliveryService()->getSyncFields() as $deliveryProperty) {
+            $value = $delivery->getOnePropertyValue($this->getProperty($deliveryProperty));
```

Comment:
each request its a query to the DB, probably would be better to collect all properties and run single request to the storage?

Reply:
Changed implementation to use `getPropertiesValues`